framework/flamingo/sessions.go

```diff
@@ -255,24 +265,68 @@ func (m *SessionModule) FlamingoLegacyConfigAlias() map[string]string {
 	}
 }
 
-func getRedisConnectionInformation(redisURL, redisHost, redisPassword string, redisDatabase int) (string, string, int) {
+func getRedisUsername(redisURL string, redisUsername string) string {
 	if redisURL == "" {
-		return redisHost, redisPassword, redisDatabase
+		return redisUsername
 	}
 
 	parsedRedisURL, err := url.Parse(redisURL)
 	if err != nil {
-		return redisHost, redisPassword, redisDatabase
+		return redisUsername
 	}
 
-	redisHostFromURL := parsedRedisURL.Host
-	if redisHostFromURL != "" {
-		redisHost = redisHostFromURL
+	redisUsernameFromURL := parsedRedisURL.User.Username()
+	if redisUsernameFromURL != "" {
+		return redisUsernameFromURL
+	}
+
+	return redisUsername
+}
+
+func getRedisPassword(redisURL string, redisPassword string) string {
+	if redisURL == "" {
+		return redisPassword
+	}
+
+	parsedRedisURL, err := url.Parse(redisURL)
+	if err != nil {
+		return redisPassword
 	}
 
 	redisPasswordFromURL, isRedisPasswordInURL := parsedRedisURL.User.Password()
 	if isRedisPasswordInURL {
-		redisPassword = redisPasswordFromURL
+		return redisPasswordFromURL
+	}
+
+	return redisPassword
+}
+
+func getRedisHost(redisURL string, redisHost string) string {
+	if redisURL == "" {
+		return redisHost
+	}
+
+	parsedRedisURL, err := url.Parse(redisURL)
+	if err != nil {
```

Comment:
Now that we are touching this, I think this is worth not being silenced.
When I configure an URL and have a typo, this secretly falls back to the host which is not wanted and searching for the error is hard